sklearn/dummy.py

```diff
@@ -118,6 +118,7 @@ def fit(self, X, y, sample_weight=None):
         self.sparse_output_ = sp.issparse(y)
 
         if not self.sparse_output_:
+            y = np.asarray(y)
```

Comment:
do we not want to have a more complex `atleast_1d` instead?

Reply:
you mean ``column_or_1d``? I honestly don't know why we would need ``atleast_1d``. We usually want scalars to be treated differently.

Reply:
there's also ``check_array(enforce_2d=False)`` which might be more suitable here?

Reply:
yep, I tend to forget that we have both `column_or_1d` and `atleast_1d`. Ideally I think I'd rather have `ensure_ndims=n` in `check_array` maybe.